src/framework/ui/input/input.component.tsx

```diff
@@ -82,52 +119,108 @@ export class Input extends React.Component<Props> {
+          ...styles.caption,
+        },
+      },
     };
   };
 
-  private renderImageElement = (style: StyleType): React.ReactElement<ImageProps> => {
-    const { icon } = this.props;
+  private renderIconElement = (style: StyleType,
+                               icon: (style: StyleType) => IconElement): IconElement | null => {
 
-    return React.cloneElement(icon(style), { key: 0 });
+    return icon ? icon(style) : null;
   };
 
-  private renderComponentChildren = (style: StyleType): React.ReactNode => {
-    const { icon } = this.props;
+  private renderTextElement = (style: StyleType, text: string): LabelElement => {
+    return (
+      <Text style={style}>{text}</Text>
+    );
+  };
 
-    const hasIcon: boolean = icon !== undefined;
+  private renderText = (style: StyleType, text: string): LabelElement | null => {
+    const hasText: boolean = text && text.length !== 0;
 
-    return [
-      hasIcon ? this.renderImageElement(style) : undefined,
-    ];
+    return hasText ? this.renderTextElement(style, text) : null;
+  };
+
+  private renderCaptionElement = (style: CaptionStyles): React.ReactElement<ViewProps> | null => {
+    const { caption, captionIcon } = this.props;
+    const icon: IconElement | null = this.renderIconElement(style.icon, captionIcon);
+    const text: LabelElement | null = this.renderText(style.text, caption);
+
+    return (
+      <View style={style.container}>
+        {icon}
+        {text}
+      </View>
+    );
   };
 
   public render(): React.ReactElement<TextInputProps> {
-    const { themedStyle, disabled, ...derivedProps } = this.props;
+    const { themedStyle, disabled, label, icon, ...restProps } = this.props;
     const componentStyle: StyleType = this.getComponentStyle(themedStyle);
-    const componentChildren: React.ReactNode = this.renderComponentChildren(componentStyle.icon);
+    const inputIcon: IconElement | null = this.renderIconElement(componentStyle.icon, icon);
+    const inputLabel: LabelElement | null = this.renderText(componentStyle.label, label);
+    const caption: React.ReactElement<ViewProps> | null =
+      this.renderCaptionElement(componentStyle.caption);
 
     return (
-      <View style={componentStyle.container}>
-        <TextInput
-          {...derivedProps}
-          editable={!disabled}
-          onFocus={this.onFocus}
-          onEndEditing={this.onEndEditing}
-          style={componentStyle.text}
-        />
-        {componentChildren}
+      <View style={componentStyle.wrapper}>
+        {inputLabel}
+        <View style={componentStyle.container}>
+          <TextInput
+            {...restProps}
+            editable={!disabled}
+            onFocus={this.onFocus}
+            onEndEditing={this.onEndEditing}
+            style={componentStyle.text}
+          />
+          {inputIcon}
+        </View>
+        {caption}
       </View>
     );
   }
 }
 
 const styles = StyleSheet.create({
+  wrapper: {
+    flexGrow: 1,
+  },
   container: {
```

Comment:
rename this to `inputContainer` and `wrapper` to `container`

Reply:
Done.